specs/monitoring_integration_service.adoc

```diff
@@ -0,0 +1,195 @@
+grafana, the user has to run the code manually every time; which is not feasible.
+Monitoring integration will be providing other functionalities such as
+
+* Handling alerts generated by grafana
+* Making dynamic configuration changes in graphite and grafana
+
+To accomplish the above, like other storage system integration pieces in tendrl,
+monitoring-integration would also require to be able to run as a daemon, managed
+via a systemd service.
+
+== Proposed change
+
+monitoring-integration component will be run as a daemon. Instead of starting each
+and every times manually, it is installed and started once as a service managed
+by systemd.
+
+To be able to run as a daemon, the component needs some additional changes.
+
+Mananger::
+monitoring-integration execution flow is started from the Manager modules. Manager
+module is the entry point for monitoring-integration, all the gevents are initialized
+and started from the Manager modules. A Manager can kill all the gevent when
+monitoring-integration service stops.
++
+When monitoring-integration starts, the Manager module will do the all configuration
+using configuration files. It will also create a separate namespace for the
+monitoring-integration. the namespace will be called "NS.monitoring"
+
+Namespace::
+A manager will create its own namespace for the monitoring-integration to separate
+its own objects from the other modules. These objects are only accessible by
+monitoring-integration.
+
+Object::
+Each tendrl component has its own set of objects for a different purpose. Each
+object have its own structure. monitoring-integration have an config and definition
+object.
+
+```
+Sample config class for monitoring-integration:
+
+class Config(objects.BaseObject):
+    internal = True
+
+    def __init__(self, config=None, *args, **kwargs):
+        self._defs = {}
+        super(Config, self).__init__(*args, **kwargs)
+
+        self.data = config or cmn_config.load_config(
+            'node-agent', "/etc/tendrl/monitoring-integration/monitoring-integration.conf.yaml")
+        self.value = '_NS/monitoring/config'
+
+
+Sample definition class for monitoring-integration:
+
+class Definition(objects.BaseObject):
+    internal = True
+
+    def __init__(self, *args, **kwargs):
+        self._defs = {}
+        super(Definition, self).__init__(*args, **kwargs)
+
+        self.data = pkg_resources.resource_string(__name__, "monitoring_integration.yaml")
+        self._parsed_defs = yaml.safe_load(self.data)
+        self.value = '_NS/monitoring/definitions'
+```
+
+Install and run:
+
+monitoring-integration will be installed in server node only. It will be
+installed and started by tendrl-ansible as a service. There is just one instance
+of monitoring-integration installed across all clusters.
+
+== Alternatives
+
+None
+
+== Data model impact:
+
+monitoring-integration configuration stored in etcd : “_NS/monitoring/config”
```

Comment:
I feel data model impact talks about new objects added as well like `Config` and `Definition`